src/common/mfu_daos.c

```diff
@@ -4136,6 +4144,26 @@ int cont_serialize_props(struct hdf5_args *hdf5,
     return rc;
 }
 
+static bool obj_is_kv_flat(daos_obj_id_t oid)
+{
+
+#if CHECK_DAOS_API_VERSION(2, 0)
+	daos_type_t type;
+
+	type = daos_obj_id2type(oid);
+	return (type == DAOS_OT_KV_HASHED || type == DAOS_OT_ARRAY_ATTR);
```

Comment:
hm, i don't think that we should check for DAOS_OT_ARRAY_ATTR here since the intent is to find out whether daos_kv_open() should be used. In the #else, we actually very that it is NOT an array.